### PR TITLE
feat(sdk-crash-detection): Set in app for SDK frames

### LINKED ISSUE #51024: SDK Crash Detection: Set in app
Set all in app for all frames to `false` except for the SDK frames after stripping the event data. 

https://github.com/getsentry/sentry/blob/95086b406dec79e6bcef45f299a3e92f727da2c0/src/sentry/utils/sdk_crashes/sdk_crash_detection.py#L58

### PR DESCRIPTION
**Pull request by philippsuess**
Set in_app to true only for SDK frames and to false for all others.

Closes #51024

### CODE REVIEW

File: tests/sentry/utils/sdk_crashes/test_event_stripper.py [hunk 1]
```diff
@@ -161,6 +161,40 @@ def test_strip_event_data_keeps_exception_mechanism(self):
...
+            stripped_event_data, "exception", "values", -1, "stacktrace", "frames"
+        )
+
+        for stripped_frame in stripped_frames[0::-1]:
+            assert stripped_frame["in_app"] is False
+
+        cocoa_sdk_frame = stripped_frames[-1]
+        assert cocoa_sdk_frame == {
+            "function": "SentryCrashMonitor_CPPException.cpp",
+            "package": "/private/var/containers/Bundle/Application/59E988EF-46DB-4C75-8E08-10C27DC3E90E/iOS-Swift.app/Frameworks/Sentry.framework/Sentry",
```
Comment (iker-barriocanal) [inline tests/sentry/utils/sdk_crashes/test_event_stripper.py:193]: What does the UUID(?) after `/Application/` refer to? May this UUID become a flaky test?

**Review by iker-barriocanal — APPROVED** [review]

File: tests/sentry/utils/sdk_crashes/test_event_stripper.py (hunk 1 above)
Comment (philipphofmann) [reply tests/sentry/utils/sdk_crashes/test_event_stripper.py:193]: Why do you think the UUID could lead to a flaky test, @iker-barriocanal?

Comment (iker-barriocanal) [reply tests/sentry/utils/sdk_crashes/test_event_stripper.py:193]: All good in that case, I didn't know where the UUID was coming from.

Comment (philipphofmann) [reply tests/sentry/utils/sdk_crashes/test_event_stripper.py:193]: Thanks for asking 🙏, @iker-barriocanal.

**Pull request merged**